src/Toolkit/Str.php

```diff
@@ -315,6 +317,19 @@ public static function date(
 			return $format->format($time ?? time());
 		}
 
+		// automatically determine the handler from global configuration
+		// if an app instance is already running
+		if ($handler === null) {
+			if ($app = App::instance(null, true)) {
+				$handler = $app->option('date.handler', 'date');
+			}
+
+			// fall back to `date` for backwards-compatibility
+			// when the method was called without app and without
+			// overriding the default handler
+			$handler ??= 'date';
```

Comment:
If you didn't write a long version on purpose, I would write a short one like this:

```suggestion
			$handler = App::instance(null, true)?->option('date.handler') ?? 'date';
```